engine/src/main/java/org/terasology/config/SystemConfig.java

```diff
@@ -16,99 +16,63 @@
 
 package org.terasology.config;
 
+import org.terasology.config.flexible.AutoConfig;
+import org.terasology.config.flexible.Setting;
+import org.terasology.config.flexible.constraints.NumberRangeConstraint;
+
 import java.util.Locale;
 import java.util.Locale.Category;
 
-public class SystemConfig {
+import static org.terasology.config.flexible.SettingArgument.constraint;
+import static org.terasology.config.flexible.SettingArgument.defaultValue;
+import static org.terasology.config.flexible.SettingArgument.type;
+
+public class SystemConfig extends AutoConfig {
     public static final String SAVED_GAMES_ENABLED_PROPERTY = "org.terasology.savedGamesEnabled";
     public static final String PERMISSIVE_SECURITY_ENABLED_PROPERTY = "org.terasology.permissiveSecurityEnabled";
 
-    private long dayNightLengthInMs;
-    private int maxThreads;
-    private int maxSecondsBetweenSaves;
-    private int maxUnloadedChunksPercentageTillSave;
-    private boolean debugEnabled;
-    private boolean monitoringEnabled;
-    private boolean writeSaveGamesEnabled;
-    private long chunkGenerationFailTimeoutInMs;
-    private String locale;
-
-    public long getDayNightLengthInMs() {
-        return dayNightLengthInMs;
-    }
-
-    public void setDayNightLengthInMs(long dayNightLengthInMs) {
-        this.dayNightLengthInMs = dayNightLengthInMs;
-    }
-
-    public int getMaxThreads() {
-        return maxThreads;
-    }
-
-    public void setMaxThreads(int maxThreads) {
-        this.maxThreads = maxThreads;
-    }
-
-    public int getMaxSecondsBetweenSaves() {
-        return maxSecondsBetweenSaves;
-    }
-
-    public void setMaxSecondsBetweenSaves(int maxSecondsBetweenSaves) {
-        this.maxSecondsBetweenSaves = maxSecondsBetweenSaves;
-    }
-
-    public int getMaxUnloadedChunksPercentageTillSave() {
-        return maxUnloadedChunksPercentageTillSave;
-    }
-
-    public void setMaxUnloadedChunksPercentageTillSave(int maxUnloadedChunksPercentageTillSave) {
-        this.maxUnloadedChunksPercentageTillSave = maxUnloadedChunksPercentageTillSave;
-    }
-
-    public boolean isDebugEnabled() {
-        return debugEnabled;
-    }
-
-    public void setDebugEnabled(boolean debugEnabled) {
-        this.debugEnabled = debugEnabled;
-    }
-
-    public boolean isMonitoringEnabled() {
-        return monitoringEnabled;
-    }
-
-    public void setMonitoringEnabled(boolean monitoringEnabled) {
-        this.monitoringEnabled = monitoringEnabled;
-    }
-
-    public boolean isWriteSaveGamesEnabled() {
-        String property = System.getProperty(SAVED_GAMES_ENABLED_PROPERTY);
```

Comment:
I don't know how important the overriding of this setting is, but if this is something we encounter frequently it might be worth looking into adding an overriding feature into `AutoConfig` or the setting UI. For now we might get away with setting this as the default value of the `Setting`, but we'll need expert opinion on how this case should be handled.

Reply:
I have some code about this locally.
Extends DSL with:
systemProperty(String, Function<String,V>)
Where function - converter,
V - settings type